mobile_verifier/src/unique_connections/ingestor.rs

```diff
@@ -0,0 +1,181 @@
+            VerifiedUniqueConnectionsIngestReportV1::file_sink(
+                settings.store_base_path(),
+                file_upload.clone(),
+                FileSinkCommitStrategy::Manual,
+                FileSinkRollTime::Default,
+                env!("CARGO_PKG_NAME"),
+            )
+            .await?;
+
+        let (unique_connections_ingest, unique_connections_server) =
+            file_source::Continuous::msg_source::<UniqueConnectionsIngestReport, _>()
+                .state(pool.clone())
+                .store(file_store.clone())
+                .lookback(LookbackBehavior::StartAfter(settings.start_after))
+                .prefix(FileType::UniqueConnectionsReport.to_string())
+                .create()
+                .await?;
+
+        let radio_threshold_ingestor = Self::new(
+            pool.clone(),
+            unique_connections_ingest,
+            verified_unique_connections,
+            authorization_verifier,
+        );
+
+        Ok(TaskManager::builder()
+            .add_task(verified_unique_conections_server)
+            .add_task(radio_threshold_ingestor)
+            .add_task(unique_connections_server)
+            .build())
+    }
+
+    pub fn new(
+        pool: PgPool,
+        unique_connections_receiver: Receiver<FileInfoStream<UniqueConnectionsIngestReport>>,
+        verified_unique_connections_sink: FileSinkClient<VerifiedUniqueConnectionsIngestReportV1>,
+        authorization_verifier: AV,
+    ) -> Self {
+        Self {
+            pool,
+            unique_connections_receiver,
+            verified_unique_connections_sink,
+            authorization_verifier,
+        }
+    }
+
+    async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("starting unique connections ingestor");
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => break,
+                Some(file) = self.unique_connections_receiver.recv() => {
+                    self.process_unique_connections_file(file).await?;
+                }
+            }
+        }
+        tracing::info!("stopping unique connections ingestor");
+        Ok(())
+    }
+
+    async fn process_unique_connections_file(
+        &self,
+        file_info_stream: FileInfoStream<UniqueConnectionsIngestReport>,
+    ) -> anyhow::Result<()> {
+        let mut txn = self.pool.begin().await?;
+
+        let mut stream = file_info_stream.into_stream(&mut txn).await?;
+        while let Some(unique_connections_report) = stream.next().await {
+            let verified_report_status = self
+                .verify_unique_connection_report(&unique_connections_report.report)
+                .await;
+
+            if matches!(
+                verified_report_status,
+                VerifiedUniqueConnectionsIngestReportStatus::Valid
+            ) {
+                db::save(&mut txn, &unique_connections_report).await?;
```

Comment:
why not do a batch insert for better performance?

Reply:
batch inserting as of [d7a1127](https://github.com/helium/oracles/pull/904/commits/d7a1127d0c57f19794b7cf868a31bafeac75b33c)